### ISSUE TITLE
Follow should have an option to follow community posts as well

### ISSUE THREAD

**Issue by lifenautjoe** [OkunaOrg/okuna-app#506]
When we follow someone's profile, we see their community posts as well if the option is enabled. Therefore we should have a way that people also can follow these contributions.

Thinking about a bottom sheet dialog that appears in the bottom after pressing follow with the following...

```
____________________
| Profile posts   X |
| Community posts X |
|                   |
(Cancel)     (Follow)
---------------------
```

We should also make sure if the person chooses no to share his community post contributions on the profile, these should as well not be shown on other people's timelines.

Perhaps we need to rephrase the whole "Show community posts in profile" option to something like "Share community posts" with a description like. "The posts you share with public communities will also be shared with your followers."

Thoughts? Ideas?

**Comment by uiboy** [comment]
If we are ever going to implement "pervasive following" like outlined above this is a good blueprint, which puts the choice ultimately in the creators hands. 

Concerns:- 
Although, while i can see the pros, im still a bit concerned that it will result in posts from active accounts hijacking timelines -> which will eventually cause people to just choose the option of following only the public posts.

Imagine me following 100 accounts, now even if 10-20 of those users post to various communities im not part of in a given timeframe, that already takes up a significant part of my timeline pushing posts from communities that im actually part of far below. 

Solution ideas:- 
We need to add filters to the timelines that go hand-in-hand with this feature, which should allow users to switch 
1. only posts from communities they are part of
2. posts to communities from users they follow.

**Comment by lifenautjoe** [comment]
Hm I think there's a misunderstanding. The only thing that would change is that the posters posts to public communities would also be shown on their followers timelines. The concept of Following would actually apply to all the public post from the person not only their Circles posts.

**Comment by lifenautjoe** [comment]
Updated comment

**Comment by uiboy** [comment]
No, i understood the way you meant it. So for eg. if I follow you, and you post to c/music .. but if im not part of c/music i still would see it on my timeline right.  This means if you post to multiple public communities, those posts will end up on my timeline. Because I chose the full follow option. That is what i meant by hijacking. 

With filters I didnt mean it will be per user, it will be like the "exclude communities im part of" toggle we have on explore tab, ie a toggle to see/or not community posts from all followers.

**Comment by lifenautjoe** [comment]
Hmmm placing myself as the poster, I see the concern now.. People would get their entire timeline loaded with my posts to several coms.. I'm not sure whether more filters would solve the issue or perhaps that people's posting habits would change due to not wanting to be unfollowed/blocked for excessively posting. Also, the issue would only arise when the subject is not interesting for the follower, so perhaps some way to hide posts from certain communities would help get rid of that case.

Let's give this some more thought...
